extensions/ffghdfh/indexedDB.js

```diff
@@ -0,0 +1,222 @@
+        const request = store.delete(key);
+
+        return new Promise((resolve, reject) => {
+          request.onsuccess = () => resolve();
+          request.onerror = () => resolve();
+        });
+      });
+    }
+
+    // Check if a key exists in the IndexedDB
+    _keyExists(key) {
+      return this._initDB().then((db) => {
+        const transaction = db.transaction(["data"], "readonly");
+        const store = transaction.objectStore("data");
+        const request = store.get(key);
+
+        return new Promise((resolve) => {
+          request.onsuccess = () => resolve(request.result !== undefined);
+        });
+      });
+    }
+
+    // Scratch blocks implementation
+    getInfo() {
+      return {
+        id: "ffghdfhIndexedDB",
+        name: "IndexedDB",
+        color1: "#fd7f54", // Main block color
+        color2: "#c26140",
+        color3: "#b35b3d", // Slightly darker outline color
+        blocks: [
+          {
+            opcode: "setDBName",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set database name [NAME]",
+            arguments: {
+              NAME: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "MyScratchDB",
+              },
+            },
+          },
+          {
+            opcode: "setValue",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set key [KEY] to [VALUE]",
+            arguments: {
+              KEY: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "myKey",
+              },
+              VALUE: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "myValue",
+              },
+            },
+          },
+          {
+            opcode: "getValue",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "get value for key [KEY]",
+            arguments: {
+              KEY: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "myKey",
+              },
+            },
+          },
+          {
+            opcode: "deleteAllKeys",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "delete all keys",
+          },
+          {
+            opcode: "deleteKey",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "delete key [KEY]",
+            arguments: {
+              KEY: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "myKey",
+              },
+            },
+          },
+          {
+            opcode: "keyExists",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "key [KEY] exists?",
+            arguments: {
+              KEY: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "myKey",
+              },
+            },
+          },
+        ],
+      };
+    }
+
+    setDBName(args) {
+      const { NAME } = args;
```

Comment:
You could do this in the function argument as opposed to setting it retroactively, if you wanted to

Reply:
uhh i'm sorry, but i'm not quite sure what you mean by that 😅

Reply:
oh i got it, one sec imma modify it